lib/webflow/client.rb

```diff
@@ -21,85 +21,104 @@ def remaining
     end
 
-
     # https://developers.webflow.com/?javascript#get-all-items-for-a-collection
     # returns json object with data to help paginate collection
     #
     # {
-    #   items:  your list of items returned,
-    #   count:  number of items returned,
-    #   limit:  the limit specified in the request (default: 100),
-    #   offset: the offset specified for pagination (default: 0),
-    #   total:  total # of items in the collection
+    #   items: [] your list of items returned,
+    #   pagination: {
+    #     limit:  the limit specified in the request (default: 100),
+    #     offset: the offset specified for pagination (default: 0),
+    #     total:  total # of items in the collection
+    #   }
     # }
     #
     # page starts at 1
     def paginate_items(collection_id, per_page: 100, page: 1)
       get("/collections/#{collection_id}/items", params: { limit: per_page, offset: per_page * (page - 1) })
     end
 
-
-    def items(collection_id, limit: 100)
+    def items(collection_id, limit: 100) # rubocop:disable Metrics/MethodLength
       fetched_items = []
       num_pages     = (limit.to_f / 100.0).ceil
-      per_page      = limit > 100 ? 100 : limit
+      per_page      = [limit, 100].min
 
       num_pages.times do |i|
-        resp = paginate_items(collection_id, per_page: per_page, page: i+1)
-        items = resp['items']
-        yield(items) if block_given?
-        fetched_items += items
-        limit -= resp['count']
-        break if limit <= 0 || resp['total'] <= fetched_items.length
+        response = paginate_items(collection_id, per_page: per_page, page: i + 1)
+        items = response.fetch(:items)
+
+        if block_given?
+          yield(items)
+        else
+          fetched_items += items
+        end
       end
 
       fetched_items
     end
 
     def item(collection_id, item_id)
-      json = get("/collections/#{collection_id}/items/#{item_id}")
-      json['items'].first
+      get("/collections/#{collection_id}/items/#{item_id}")
+    end
+
+    def create_item(collection_id, data, is_archived: false, is_draft: false, publish: false)
+      result = post("/collections/#{collection_id}/items",
+                    { isArchived: is_archived, isDraft: is_draft, fieldData: data })
+      return result unless publish
+
+      publish_item(collection_id, result.fetch(:id))
+      result
+    end
+
+    def update_item(collection_id, item_id, data, is_archived: false, is_draft: false, publish: false) # rubocop:disable Metrics/ParameterLists
+      result = patch("/collections/#{collection_id}/items/#{item_id}",
+                     { isArchived: is_archived, isDraft: is_draft, fieldData: data }.compact)
+      return result unless publish
+
+      publish_item(collection_id, item_id)
+      result
     end
 
-    def create_item(collection_id, data, live: false)
-      post("/collections/#{collection_id}/items", {fields: data}, live: live)
+    def delete_item(collection_id, item_id)
```

Comment:
`delete_and_unpublish` then?

Reply:
If it's deleted, it's pretty much unpublished, since it's actually gone. This is Webflow API internal bug that we cannot fix. And this is an implementation detail of how `delete_item` works. I don't think the user of webflow-ruby needs to know all the workarounds we need to implement. They just want to know the API & methods of the gem so that they can use them. `delete_item` to me sounds much clearer than `delete_and_unpublish`.

Reply:
yeah, that's probably true. as long as the underlying primites are still exposed and one can implement something more custom i'm fine with it.

Reply:
Ahhhh, I see...
I wasn't keeping in consideration supporting the underlying low-level API access.

In this case, we lost the original "delete item" low-level API access.
Since it wasn't working as expected in the first place, I'm not sure if we really lost anything?
I "lost" a lot of time debugging and figuring out why items with `publish: true` (ex `live: true`) aren't visible on the live site. The (still not fixed) Webflow bug causes changes to not being published anymore for _any_ updates to _any_ items, after the buggy "delete item" request is made.

I'm just thinking of the use-case where someone might need the low-level access to the API? Perhaps if there's more functionality added to the API and the gem doesn't get updated regularly / on time.

Should we just expose the access to `request` and add a comment saying that it's a low-level API and there most likely exists a higher-level method that can be used instead?

Or if you want to add low-level (buggy) `delete_item`, I'd vote for keeping the working, higher-level `delete_item` as-is and add a new method `delete_item_single` / `delete_item_raw` / `delete_item_something` with the comment explaining why there are two methods and which one works / is recommended to be used.

What do you think?

Reply:
just re-reading the documentation and your comments, i think this should definitively be an additional method. 

is it correct that `delete_item` works fine if you publish the site afterwards? bc, that is what i would have expected in the first place.

Reply:
Publishing a website is discouraged and has a special very low limit of 1 call every 60 seconds:
https://developers.webflow.com/changelog/publish-site-api-new-rate-limits

It's also not the same as publishing a single item (whether it's create, update or delete). Publishing, publishes _all_ of the pending work in design + pending collection items.

But you're right that we are missing "mark for deletion" (delete, pending).
It's tricky to test delete pending as I'm not sure whether it would work or the mentioned bug would happen. The issue is that, if I remember correctly, the API returns everything as it should be, but on your actual Webflow site, you can see that the new changes are not being applied. So it's impossible to write an automated test to test this (the automated test would have to navigate to a Webflow site and check that there's no such item on the page).

If you have time please take over and feel free to look into this and add the missing functionality.
Otherwise, I can add a comment about delete pending missing in the gem and invite people to add it if someone needs it. Unfortunately, I don't have the time right now to look into this buggy functionality and try to figure out in which cases it works in which cases it doesn't.

Reply:
i dont really understand the webflow API versioning here, is the "v2 BETA" before or after "v2"? well at least there is another method for delete and publish https://docs.developers.webflow.com/v2.0.0-beta/reference/delete-item-live

Reply:
I _think_ it started as API v2 beta and then they released API v2 and "forgot" to update API v2 beta to API beta or API v3 beta.

I totally missed that they've added these `live` methods for create, update, and delete. I assume this will be a part of API v3.

When they launch API v3, I'll try to keep the gem updated.